### ISSUE TITLE
Boolean in metadata at defaults read as string

### ISSUE THREAD

**Issue by jzeneto** [jgm/pandoc#6349]
Pandoc version: 2.9.2

Hi! According to [this comment](https://github.com/jgm/pandoc/issues/6250#issuecomment-609454894), a boolean metadata specified in defaults file should be parsed as YAML boolean. But trying to configure pandoc-crossref in defaults file raises a type error.

My defaults file, `def.yaml`:

```yaml
filters:
  - pandoc-crossref

metadata:
  numberSections: true
```

An input file for testing:

```markdown
# A title

Some text.
```

Running `pandoc -d def input.txt` raises this:

    pandoc-crossref: Expected bool in metadata field numberSections but got string
    CallStack (from HasCallStack):
      error, called at lib/Text/Pandoc/CrossRef/Util/Meta.hs:73:25 in pandoc-crossref-0.3.6.2-inplace:Text.Pandoc.CrossRef.Util.Meta
    Error running filter pandoc-crossref:
    Filter returned error status 1

Is this error expected?

**Comment by jzeneto** [comment]
Although error was raised by pandoc-crossref, the error relates to a metadata field received from pandoc, which is why I opened this issue here.

**Comment by jgm** [comment]
See #6328

**Comment by jzeneto** [comment]
Thank you, @jgm! A workaround for this problem is using a dedicated metadata file just for configuring pandoc-crossref. I tested this way and it works!

**Comment by jgm** [comment]
I believe this has been fixed by #6328.
